ui/src/app/views/cohort-edit/component.spec.ts

```diff
@@ -0,0 +1,262 @@
+import {Component, DebugElement} from '@angular/core';
+import {TestBed, async, fakeAsync, tick, ComponentFixture} from '@angular/core/testing';
+import {Title, By} from '@angular/platform-browser';
+import {FormsModule} from '@angular/forms';
+import {ActivatedRoute, UrlSegment} from '@angular/router';
+import {CohortEditComponent} from 'app/views/cohort-edit/component';
+import {RouterTestingModule} from '@angular/router/testing';
+import {BehaviorSubject} from 'rxjs/BehaviorSubject';
+import {SignInService} from 'app/services/sign-in.service';
+import {WorkspaceComponent} from 'app/views/workspace/component';
+
+import {Cohort, CohortListResponse} from 'generated';
+import {CohortsService} from 'generated';
+import {Observable} from 'rxjs/Observable';
+import {Observer} from 'rxjs/Observer';
+
+// TODO: Replace with Swagger-generated model
+class Workspace {
+  id: string;
+  namespace: string;
+  cohorts: Cohort[];
+}
+
+class Context {
+  fixture: ComponentFixture<CohortEditComponent>;
+  route: UrlSegment[];
+  cohortsService: CohortsService;
+  nameField: DebugElement;
+  descriptionField: DebugElement;
+  addButton: DebugElement;
+  saveButton: DebugElement;
+
+  constructor(testBed: typeof TestBed) {
```

Comment:
TypeScript lets you create fields at the same time as declaring constructor args (hooray!). https://www.stevefenton.co.uk/2013/04/stop-manually-assigning-typescript-constructor-parameters/ The example is for private fields, but I assume it'll work for public ones too.